### PR TITLE
Add indices and restart app upon crash

### PR DESCRIPTION
**Pull request by xdzurman**
add indexes to speed up initial sync of the tangata-manu instance on Shelley testnet. Rudimentary profiling of postgres requests while syncing an instance from zero revealed that requests like these below happen frequently while syncing and without indexes they take up too much CPU and are too slow:

```
SELECT tx_hash FROM transient_snapshots WHERE (block_num = (SELECT MAX(block_num) FROM transient_snapshots)) AND (status = 'Pending') UNION (SELECT hash FROM txs WHERE (tx_state = 'Pending') AND (NOT EXISTS (SELECT 1 FROM transient_snapshots WHERE (tx_hash = hash) AND (block_num = (SELECT MAX(block_num) FROM transient_snapshots)))))

 SELECT tx_hash FROM transient_snapshots WHERE (block_num = (SELECT MAX(block_num) FROM transient_snapshots)) AND (status = 'Pending') UNION (SELECT hash FROM txs WHERE (tx_state = 'Pending') AND (NOT EXISTS (SELECT 1 FROM transient_snapshots WHERE (tx_hash = hash) AND (block_num = (SELECT MAX(block_num) FROM transient_snapshots)))))

 SELECT hash FROM txs WHERE (tx_state = 'Failed') AND (NOT EXISTS (SELECT * FROM transient_snapshots WHERE (status = 'Failed') AND (tx_hash = hash)))

 SELECT tx_hash FROM transient_snapshots WHERE (block_num = (SELECT MAX(block_num) FROM transient_snapshots)) AND (status = 'Pending') UNION (SELECT hash FROM txs WHERE (tx_state = 'Pending') AND (NOT EXISTS (SELECT 1 FROM transient_snapshots WHERE (tx_hash = hash) AND (block_num = (SELECT MAX(block_num) FROM transient_snapshots)))))

 INSERT INTO blocks (block_hash, epoch, slot, block_height, slot_leader) VALUES ('1f3784f02026fd88335bbda738e2b112b27ae6a7c75c27a5560e043240a1d264', 18, 30144, 61213, 'bb719b8438983af6eae0930ec88951081a1c6225c99e9149f4233846e9f2d57c'), ('6765d4c29666435bfa7e52f99773c7786f540a2f8b8681adbd2a7e6b31d703fe', 18, 30145, 61214, '8c92fb7b01d78e9974d3a146ac144597303dc6419cf90062456deb8140e3a81b'), ('ddbe3a9655cd3f43151f982875a8f27f3a3461272b1c54aa163cabf3131dc1cf', 18, 30151, 61215, '7186b11017e877329798ac925480585208516c4e5c30b69e38f0b997e7b72a83'), ('810c2453411482a0bbe2c4451804dfb98b84a80378b9b0d5ac4732fad40a84a2', 18, 30154, 61216, 'b0741d28cc5e2bddbcf2387973e4a574f5360671c0be1a3e4b778fad6dd7c255'), ('6b75b2c2069358e72d137b46a5317897c3bca907bba8e468cdb7f0ab4c91d834', 18, 30161, 61217, 'a19af49ed88574bd181022c38904d76482be0e57778f3ee28a6abf3769d6ac46'), ('0b1eb84cface04ce1496ec0c655f3e8a7938809e02dd92ff93a6aca8e30a4357', 18, 30177, 61218, '87421b5400a3391c2db836c81ffaad2f32b5b13e8004506dd9bcca740439c327')

 INSERT INTO tx_addresses (tx_hash, address) VALUES ('c2ff035047fffdd3edb6fee7002941fa8f4a0594458699a8f2e662b7876cbbc6', '8480c78d9ba43df0bb468c8d53e87404dd40bc1649b04ec8ce004bad2e3fb7ad4116d56c28bf1a50474e647ff00eb57291924fa4c38c9e7c45e015088387c842e4'), ('c2ff035047fffdd3edb6fee7002941fa8f4a0594458699a8f2e662b7876cbbc6', '844e6b756cc305170eb8ab588c5a1135dda11c842ff583eae1a69f87935fb8e02916d56c28bf1a50474e647ff00eb57291924fa4c38c9e7c45e015088387c842e4') ON CONFLICT DO NOTHING
```

Another improvement is extending yarn dev command with auto-restart

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,7 +8,7 @@
     "migrate-shelley": "NODE_ENV=${NODE_ENV:-develop} node-pg-migrate -m migrations-shelley",
     "build": "babel src -d dist",
     "eslint": "eslint --ignore-path .gitignore src",
-    "dev": "rimraf ./dist && yarn run build && export NODE_ENV=${NODE_ENV:-develop} && nodemon ./src/index.js --max_old_space_size=4096 --inspect --exec babel-node | bunyan",
+    "dev": "rimraf ./dist && yarn run build && export NODE_ENV=${NODE_ENV:-develop} && nodemon -x 'babel-node ./src/index.js --max_old_space_size=4096 --inspect=9229 || touch ./src/index.js' | bunyan",
```
Comment (refi93) [inline package.json:11]: @xdzurman can you explain the meaning of `--inspect==9229`?

Comment (xdzurman) [reply package.json:11]: --inspect automatically assigns port 9229 when you try to watch your application and therefore falls into conflicts with other applications or instances of the same application when you try to run them. Having the port specified saves time analyzing what went wrong and allows a quick change for another port. This problem disappears with dockerization though.

Comment (refi93) [reply package.json:11]: I see, thanks for the explanation!

Comment (xdzurman) [reply package.json:11]: After further research, another option is specifying --inspect=0, which picks a random port. This way we could potentially confuse the user, so having this deterministic approach seems better to me

**Pull request merged**